src/Listener/AddDiscussionLockedAttributes.php

```diff
@@ -20,6 +20,8 @@ public function handle(Serializing $event)
     {
         if ($event->isSerializer(DiscussionSerializer::class)) {
             $event->attributes['isLocked'] = (bool) $event->model->is_locked;
+            $event->attributes['canReply'] = $event->attributes['isLocked'] === true
```

Comment:
Although I like this proposal, it doesn't take into account that `canReply` was set to false by any other extension before this code is executed.

Reply:
Thanks , I agree but there is no way to prevent it or any other attributes when there is multiple listeners/extensions.

Reply:
Thank you :) added check for `canReply`